src/interpreter.cr

```diff
@@ -0,0 +1,95 @@
+
+struct Interpreter
+  @@shell : String | Nil
+
+  protected def self.shell : String
+    {% if flag?(:darwin) %}
+      @@shell ||= begin
+        Process.find_executable(ENV.fetch("RUBYSHELL", "dash")).as(String)
+      rescue TypeCastError
+        Process.find_executable(ENV.fetch("RUBYSHELL", "sh")).as(String)
```

Comment:
No point in using `RUBYSHELL` again here, especially if it caused the TypeCastError.

Reply:
Oh true. The `TypeCastError` it would be catching is if `find_executable` fails to find `dash` (_basically, pre-Catalina_). Of course, that would mean `RUBYSHELL` is empty, so definitely missed that one lol